### PR TITLE
docs: Fix 'config' mention in 'hash' docs

### PR DESCRIPTION
**Pull request by travier**

### CODE REVIEW

**Comment by travier** [comment]
How do we feel about adding a note like "If compression is set, then the hash is for the decompressed content" to all of those instances for https://github.com/coreos/ignition/issues/961 ?

**Comment by bgilbert** [comment]
Let's aim to keep the Butane spec docs in sync with the Ignition ones.  The Ignition spec docs opted for "the hash of the contents", which is a little clearer in the context of the spec.

"If `compression` is specified, the hash describes the decompressed contents" SGTM.

**Comment by bgilbert** [comment]
Also, we typically release-note changes to the spec docs, even if they're minor clarifications.

**Comment by bgilbert** [comment]
@travier, are you planning to handle updating this?

**Comment by travier** [comment]
I need to update it and sync it with the Ignition docs. Open to someone else picking it up.

File: docs/config-fcos-v1_0.md [hunk 1]
```diff
@@ -64,13 +64,13 @@ The Fedora CoreOS configuration is a YAML document conforming to the following s
       * **_source_** (string): the URL of the file contents. Supported schemes are `http`, `https`, `tftp`, `s3`, and [`data`][rfc2397]. When using `http`, it is advisable to use the verification option to ensure the contents haven't been modified. If source is omitted and a regular file already exists at the path, Ignition will do nothing. If source is omitted and no file exists, an empty file will be created. Mutually exclusive with `inline`.
       * **_inline_** (string): the contents of the file. Mutually exclusive with `source`.
       * **_verification_** (object): options related to the verification of the file contents.
-        * **_hash_** (string): the hash of the config, in the form `<type>-<value>` where type is `sha512`.
+        * **_hash_** (string): the hash of the contents, in the form `<type>-<value>` where type is `sha512`. If compression is specified, the hash describes the decompressed contents.
```
Comment (bgilbert) [inline docs/config-fcos-v1_0.md:67]:
```suggestion
        * **_hash_** (string): the hash of the contents, in the form `<type>-<value>` where type is `sha512`. If `compression` is specified, the hash describes the decompressed contents.
```

File: docs/release-notes.md [hunk 2]
```diff
@@ -22,6 +22,8 @@ nav_order: 9
 - Drop `LuksOption` and `RaidOption` types _(Go API for fcos 1.5.0-experimental,
   flatcar 1.1.0-experimental, openshift 4.13.0-experimental)_
 - Require Go 1.18+
+- Correctly document `hash`, and clarify how its effected by the `compression`
```
Comment (bgilbert) [inline docs/release-notes.md:25]: This should be under `Docs changes`.

How about:

```
- Document that `hash` fields describe decompressed contents
- Clarify spec docs for `files`/`luks` `hash` fields
```

**Review by bgilbert — COMMENTED** [review]
We'll need the "decompressed contents" note on the other `hash` fields too.  Ignition spec docs will need a similar update.

File: docs/release-notes.md (hunk 2 above)
Comment (prestist) [reply docs/release-notes.md:25]: Sigh, I should have made sure there was not a doc's section. Sorry about that. And yes those sound better.
Ill make a follow up pr for ignition as well.

File: docs/release-notes.md [hunk 3]
```diff
@@ -25,6 +25,8 @@ nav_order: 9
 
 ### Docs changes
 
+- Document that `hash` fields describe decompressed contents/config
```
Comment (bgilbert) [inline docs/release-notes.md:28]: Maybe:
```suggestion
- Document that `hash` fields describe decompressed data
```

**Review by bgilbert — COMMENTED** [review]
The `certificate_authorities` `hash` fields generally need to add the sentence about `compression`.

File: docs/release-notes.md (hunk 3 above)
Comment (prestist) [reply docs/release-notes.md]: Ah, ok, and added the change for the `certificate_authorities`

File: docs/config-openshift-v4_12.md [hunk 4]
```diff
@@ -93,7 +93,7 @@ The OpenShift configuration is a YAML document conforming to the following speci
       * **_inline_** (string): the contents of the file. Mutually exclusive with `source` and `local`.
       * **_local_** (string): a local path to the contents of the file, relative to the directory specified by the `--files-dir` command-line argument. Mutually exclusive with `source` and `inline`.
       * **_verification_** (object): options related to the verification of the file contents.
-        * **_hash_** (string): the hash of the config, in the form `<type>-<value>` where type is either `sha512` or `sha256`.
+        * **_hash_** (string): the hash of the config, in the form `<type>-<value>` where type is either `sha512` or `sha256`. If `compression` is specified, the hash describes the decompressed config.
```
Comment (bgilbert) [inline docs/config-openshift-v4_12.md:96]:
```suggestion
        * **_hash_** (string): the hash of the contents, in the form `<type>-<value>` where type is either `sha512` or `sha256`. If `compression` is specified, the hash describes the decompressed contents.
```

File: docs/config-openshift-v4_12.md [hunk 5]
```diff
@@ -113,7 +113,7 @@ The OpenShift configuration is a YAML document conforming to the following speci
         * **name** (string): the header name.
         * **_value_** (string): the header contents.
       * **_verification_** (object): options related to the verification of the file contents.
-        * **_hash_** (string): the hash of the config, in the form `<type>-<value>` where type is either `sha512` or `sha256`.
+        * **_hash_** (string): the hash of the config, in the form `<type>-<value>` where type is either `sha512` or `sha256`. If `compression` is specified, the hash describes the decompressed config.
```
Comment (bgilbert) [inline docs/config-openshift-v4_12.md:116]:
```suggestion
        * **_hash_** (string): the hash of the contents, in the form `<type>-<value>` where type is either `sha512` or `sha256`. If `compression` is specified, the hash describes the decompressed contents.
```

File: docs/config-r4e-v1_0.md [hunk 6]
```diff
@@ -64,7 +64,7 @@ The RHEL for Edge configuration is a YAML document conforming to the following s
         * **name** (string): the header name.
         * **_value_** (string): the header contents.
       * **_verification_** (object): options related to the verification of the file contents.
-        * **_hash_** (string): the hash of the config, in the form `<type>-<value>` where type is either `sha512` or `sha256`.
+        * **_hash_** (string): the hash of the config, in the form `<type>-<value>` where type is either `sha512` or `sha256`. If `compression` is specified, the hash describes the decompressed config.
```
Comment (bgilbert) [inline docs/config-r4e-v1_0.md:67]:
```suggestion
        * **_hash_** (string): the hash of the contents, in the form `<type>-<value>` where type is either `sha512` or `sha256`. If `compression` is specified, the hash describes the decompressed contents.
```

File: docs/config-r4e-v1_0.md [hunk 7]
```diff
@@ -74,7 +74,7 @@ The RHEL for Edge configuration is a YAML document conforming to the following s
         * **name** (string): the header name.
         * **_value_** (string): the header contents.
       * **_verification_** (object): options related to the verification of the appended contents.
-        * **_hash_** (string): the hash of the config, in the form `<type>-<value>` where type is either `sha512` or `sha256`.
+        * **_hash_** (string): the hash of the config, in the form `<type>-<value>` where type is either `sha512` or `sha256`. If `compression` is specified, the hash describes the decompressed config.
```
Comment (bgilbert) [inline docs/config-r4e-v1_0.md:77]:
```suggestion
        * **_hash_** (string): the hash of the contents, in the form `<type>-<value>` where type is either `sha512` or `sha256`. If `compression` is specified, the hash describes the decompressed contents.
```

File: docs/config-r4e-v1_1-exp.md [hunk 8]
```diff
@@ -66,7 +66,7 @@ The RHEL for Edge configuration is a YAML document conforming to the following s
         * **name** (string): the header name.
         * **_value_** (string): the header contents.
       * **_verification_** (object): options related to the verification of the file contents.
-        * **_hash_** (string): the hash of the config, in the form `<type>-<value>` where type is either `sha512` or `sha256`.
+        * **_hash_** (string): the hash of the config, in the form `<type>-<value>` where type is either `sha512` or `sha256`. If `compression` is specified, the hash describes the decompressed config.
```
Comment (bgilbert) [inline docs/config-r4e-v1_1-exp.md:69]:
```suggestion
        * **_hash_** (string): the hash of the contents, in the form `<type>-<value>` where type is either `sha512` or `sha256`. If `compression` is specified, the hash describes the decompressed contents.
```

File: docs/config-r4e-v1_1-exp.md [hunk 9]
```diff
@@ -76,7 +76,7 @@ The RHEL for Edge configuration is a YAML document conforming to the following s
         * **name** (string): the header name.
         * **_value_** (string): the header contents.
       * **_verification_** (object): options related to the verification of the appended contents.
-        * **_hash_** (string): the hash of the config, in the form `<type>-<value>` where type is either `sha512` or `sha256`.
+        * **_hash_** (string): the hash of the config, in the form `<type>-<value>` where type is either `sha512` or `sha256`. If `compression` is specified, the hash describes the decompressed config.
```
Comment (bgilbert) [inline docs/config-r4e-v1_1-exp.md:79]:
```suggestion
        * **_hash_** (string): the hash of the contents, in the form `<type>-<value>` where type is either `sha512` or `sha256`. If `compression` is specified, the hash describes the decompressed contents.
```

**Review by bgilbert — APPROVED** [review]

**Comment by travier** [comment]
Thanks @prestist !

**Pull request merged**